### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by bootstraponline**
edo_warnings

### CODE REVIEW

**Comment by ynzhang0509** [comment]
Looks like this change is not compatible with Xcode 9

**Comment by bootstraponline** [comment]
We're using Xcode 10.1. How important is legacy Xcode support? Usually iOS teams are good about upgrading to the latest stable release

**Comment by tirodkar** [comment]
I can just upgrade the Xcode to 10.1 in bitrise to see what warnings surface. Thanks a lot for this @bootstraponline.

**Comment by wuhao5** [comment]
> We're using Xcode 10.1. How important is legacy Xcode support? Usually iOS teams are good about upgrading to the latest stable release

can you update the travis.yml to only run with xcode 10?

File: eDistantObject.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 46;
+	objectVersion = 50;
```
Comment (wuhao5) [inline eDistantObject.xcodeproj/project.pbxproj]: can you revert this?

File: eDistantObject.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -965,7 +965,7 @@
 		C845024D20DD9CB900D7350F /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 0940;
+				LastUpgradeCheck = 1010;
```
Comment (wuhao5) [inline eDistantObject.xcodeproj/project.pbxproj]: also revert this.

File: eDistantObject.xcodeproj/xcshareddata/xcschemes/TravisTests.xcscheme [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0940"
+   LastUpgradeVersion = "1010"
```
Comment (wuhao5) [inline eDistantObject.xcodeproj/xcshareddata/xcschemes/TravisTests.xcscheme]: and this.

**Comment by bootstraponline** [comment]
Travis is RIP. I think we should standardize on Bitrise.

https://news.ycombinator.com/item?id=19218036

Travis has always been painful on mobile, now it'll never get any better. If I get this working on Bitrise, is it necessary to revert the changes still? Xcode makes them automatically so undoing them for each commit seems a bit painful.

**Comment by wuhao5** [comment]
> Travis is RIP. I think we should standardize on Bitrise.
> 
> https://news.ycombinator.com/item?id=19218036
> 
> Travis has always been painful on mobile, now it'll never get any better. If I get this working on Bitrise, is it necessary to revert the changes still? Xcode makes them automatically so undoing them for each commit seems a bit painful.

we could evaluate that but right now we are relying on Travis CI for all the submissions.

**Comment by wuhao5** [comment]
To be more precise, anything that breaks Travis may risk being reverted by our automation system.

**Comment by bootstraponline** [comment]
I will work on fixing the CI process for eDistantObject

**Comment by bootstraponline** [comment]
I setup the bitrise job here:
https://app.bitrise.io/app/29ae3acccf8b3a0f#/workflow

I pushed a new commit with the requested revisions.

**Comment by wuhao5** [comment]
> I setup the bitrise job here:
> https://app.bitrise.io/app/29ae3acccf8b3a0f#/workflow
> 
> I pushed a new commit with the requested revisions.

awesome, thanks!

**Pull request merged**